### PR TITLE
RewriteLock support

### PR DESCRIPTION
**Pull request by wickedOne**
as addition to the [RewriteMap](https://github.com/puppetlabs/puppetlabs-apache/pull/900) feature i've added the option to define a [RewriteLock](http://httpd.apache.org/docs/2.2/mod/mod_rewrite.html#RewriteLock) for apache version <= 2.2 which is considered best practice when using a RewriteMap of type [prg](https://httpd.apache.org/docs/2.2/rewrite/rewritemap.html#prg)

### CODE REVIEW

**Comment by igalic** [comment]
Thanks for catching this oversight! would you mind also adding a test and a doc entry?

**Comment by wickedOne** [comment]
yes will do

**Comment by wickedOne** [comment]
done…

File: manifests/init.pp [hunk 1]
```diff
@@ -55,6 +55,7 @@
   $conf_template          = $::apache::params::conf_template,
   $servername             = $::apache::params::servername,
   $pidfile                = $::apache::params::pidfile,
+  $rewrite_lock           = false,
```
Comment (igalic) [inline manifests/init.pp]: let's make this `undef` by default, since we're dealing with a `String` (path) here.

File: manifests/init.pp [hunk 2]
```diff
@@ -317,6 +318,7 @@
     # - $server_tokens
     # - $server_signature
     # - $trace_enable
+    # - $rewrite_lock
```
Comment (igalic) [inline manifests/init.pp]: we should [validate this variable](https://github.com/puppetlabs/puppetlabs-stdlib#validate_absolute_path),
```puppet
if $rewrite_lock {
  validate_absolute_path($rewrite_lock)
}
```

File: spec/classes/apache_spec.rb [hunk 3]
```diff
@@ -388,6 +388,37 @@
         it { is_expected.to contain_file("/etc/httpd/conf/httpd.conf").with_content %r{^IncludeOptional "/etc/httpd/conf\.d/\*\.conf"$} }
       end
 
+      context "with Apache version < 2.4" do
+        let :params do
+          {
+            :apache_version => '2.2',
+            :rewrite_lock => '/etc/httpd/RewriteLock'
```
Comment (igalic) [inline spec/classes/apache_spec.rb]: shouldn't this be somewhere where httpd has write acces?
(i sure hope you're not giving it write access to its own config ;)

**Comment by igalic** [comment]
looks very good otherwise, thanks @wickedOne!

**Comment by wickedOne** [comment]
thanks for pointing all that out @igalic 
made changes accordingly and ammended them to my last commit.

moved the lockfile to location ``/var/lock/subsys/rewrite-lock`` in ``spec/classes/apache_spec.rb`` as i wasn't able to find something of a best practice for this specific lockfile and as that's where the other lockfiles are stored… hope that's ok with you

**Comment by igalic** [comment]
\o/ @wickedOne!

**Comment by wickedOne** [comment]
thanx @igalic for the super quick merge! \,,/.

**Pull request merged**